Sources/ArgumentParser/Parsable Properties/CompletionKind.swift

```diff
@@ -9,56 +9,225 @@
+    /// Swift Argument Parser does not perform any escaping or quoting on the
+    /// given shell command.
+    ///
+    /// The given shell command is included in a completion script when it is
+    /// generated.
     case shellCommand(String)
 
-    /// Generate completions using the given closure.
+    /// The completion candidates are the elements of the `[String]` returned by
+    /// the given closure when it is run when a user requests completions.
+    ///
+    /// Completion candidates are interpreted by the requesting shell as
+    /// literals. They must be neither escaped nor quoted; Swift Argument Parser
+    /// escapes or quotes them as necessary for the requesting shell.
+    ///
+    /// The given closure is evaluated after a user invokes completion in their
+    /// shell (normally by pressing TAB); it is not evaluated when a completion
+    /// script is generated.
+    ///
+    /// Depending on which shell is requesting completions, the `[String]`
+    /// argument passed to the given closure contains:
+    ///
+    /// - the command being called as the first element (bash & fish, not zsh)
+    /// - either:
+    ///   - all the arguments on the command line (bash & zsh, presumably fish
+    ///     4+)
+    ///   - the arguments on the command line only up to & including the
+    ///     argument for which completions are being requested (fish 3-)
+    ///
+    /// The arguments are passed to Swift verbatim, not unquoted. e.g., the
+    /// representation in Swift of the shell argument `"abc\\""def"` would be
+    /// exactly the same, including the quotes & the double backslash.
+    ///
+    /// Due to limitations in fish 3, arguments are passed to Swift unquoted.
+    /// e.g., the aforementioned example argument would be passed as `abc\def`.
+    /// fish 4 will not be constrained by such limitations, so this behavior
+    /// will be able to be fixed in a subsequent update to Swift Argument
+    /// Parser.
     case custom(@Sendable ([String]) -> [String])
   }
 
   internal var kind: Kind
 
-  /// Use the default completion kind for the value's type.
+  /// Use the default completion kind for the argument's or option value's type.
   public static var `default`: CompletionKind {
     CompletionKind(kind: .default)
   }
 
-  /// Use the specified list of completion strings.
+  /// The completion candidates are the elements of `words`.
+  ///
+  /// Completion candidates are interpreted by the requesting shell as literals.
+  /// They must be neither escaped nor quoted; Swift Argument Parser escapes or
+  /// quotes them as necessary for the requesting shell.
+  ///
+  /// The completion candidates are included in a completion script when it is
+  /// generated.
   public static func list(_ words: [String]) -> CompletionKind {
     CompletionKind(kind: .list(words))
   }
 
-  /// Complete file names.
+  // swift-format-ignore: BeginDocumentationCommentWithOneLineSummary
+  /// The completion candidates include:
```

Comment:
The one-line summary is what shows up as an abstract when an API is included as a list, so this would just be a sentence fragment. Let's use a full sentence ("...directory and file names, filtered by the given list of extensions"?) and then include the detail about exactly which file names in the body of the doc comment.